gossipsubimpl.go

```diff
@@ -0,0 +1,23 @@
+package pubsub
+
+import (
+	"context"
+
+	host "github.com/libp2p/go-libp2p-host"
+	protocol "github.com/libp2p/go-libp2p-protocol"
+)
+
+const (
+	GossipSubID = protocol.ID("/meshsub/1.0.0")
+)
+
+// NewGossipSub returns a new PubSub object using a GossipConfigurableRouter as the router and configured with classic GossipSub parameters.
+func NewGossipSub(ctx context.Context, h host.Host, opts ...Option) (*PubSub, error) {
```

Comment:
It feels like this should be in a separate file now that NewGossipSub is just one of multiple possible gossip based configurations. Up for any suggestions on where this code lives or what to name the file.